src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Completion/DirectiveAttributeCompletionItemProvider.cs

```diff
@@ -178,18 +181,43 @@ bool TryAddCompletion(string attributeName, BoundAttributeDescriptor boundAttrib
 
             void AddCompletion(string attributeName, BoundAttributeDescriptor boundAttributeDescriptor, TagHelperDescriptor tagHelperDescriptor)
             {
-                if (!attributeCompletions.TryGetValue(attributeName, out var attributeDescriptionInfos))
+                if (!attributeCompletions.TryGetValue(attributeName, out var attributeDetails))
                 {
-                    attributeDescriptionInfos = new HashSet<AttributeDescriptionInfo>();
-                    attributeCompletions[attributeName] = attributeDescriptionInfos;
+                    attributeDetails = (new HashSet<AttributeDescriptionInfo>(), new HashSet<string>());
+                    attributeCompletions[attributeName] = attributeDetails;
                 }
 
+                (var attributeDescriptionInfos, var commitCharacters) = attributeDetails;
+
                 var descriptionInfo = new AttributeDescriptionInfo(
                     boundAttributeDescriptor.TypeName,
                     tagHelperDescriptor.GetTypeName(),
                     boundAttributeDescriptor.GetPropertyName(),
                     boundAttributeDescriptor.Documentation);
                 attributeDescriptionInfos.Add(descriptionInfo);
+
+                if (attributeName.EndsWith("..."))
+                {
+                    // Indexer attribute, we don't want to commit with standard chars
+                    return;
+                }
+
+                if (!commitCharacters.Contains("="))
+                {
+                    commitCharacters.Add("=");
+                }
+
+                if (!commitCharacters.Contains(" ") &&
+                    tagHelperDescriptor.BoundAttributes.Any(b => b.IsBooleanProperty))
```

Comment:
Is this also true for Indexer attributes?

Reply:
Indexer attributes would've been filtered out https://github.com/dotnet/aspnetcore-tooling/pull/1825/files/a52fff862fa5b76380e4ee0721af3df6ab23a729#diff-f28100f7068e848f19fb6ced2a46e4a3R199

We aren't offering any commit chars for indexer attributes (having ` ` available would mean a completion of `bind- ` for instance)